page-block-editor.php

```diff
@@ -24,14 +24,26 @@
 	);
 
 	if ( $show_title ) {
-		$parent_page = wp_get_post_parent_id( get_the_ID() );
-
 		$template_args = array(
-			'h4_link'  => ! empty( $parent_page ) ? get_the_permalink( $parent_page ) : '',
-			'h4_title' => ! empty( $parent_page ) ? get_the_title( $parent_page ) : '',
 			'h1_title' => get_the_title(),
 		);
 
+		$breadcrumb_link_switch = get_post_meta( get_the_ID(), 'show_breadcrumb_links', true );
+		if ( $breadcrumb_link_switch ) {
+			$parent_page = wp_get_post_parent_id( get_the_ID() );
+
+			$breadcrumb_link_custom_title = get_post_meta( get_the_ID(), 'breadcrumb_link_title', true );
+			$breadcrumb_link_title = ( ! empty( $breadcrumb_link_custom_title ) ) ? $breadcrumb_link_custom_title : get_the_title( $parent_page );
+
+			$breadcrumb_link_custom_url = get_post_meta( get_the_ID(), 'breadcrumb_link_url', true );
+			$breakcrumb_link = ( ! empty( $breadcrumb_link_custom_url ) ) ? $breadcrumb_link_custom_url : get_the_permalink( $parent_page );
+
+			$template_args = array(
+				'h4_link'  => $breakcrumb_link,
```

Comment:
We've already defined `$template_args` above, so recreating the array will wipe out the previous `h1_title` property. Should we be adding properties here, instead?

e.g.,

```php
$template_args['h4_link'] = $breadcrumb_link;
```